packages/block-library/src/group/deprecated.js

```diff
@@ -42,6 +42,139 @@ const migrateAttributes = ( attributes ) => {
 };
 
 const deprecated = [
+	// Version with preset gradient color and background image.
```

Comment:
A note on this. 

Merging color gradient and background image is technically a migration, not a deprecation. Custom blocks might need this as well. 

Furthermore, if migrating `color.gradient` from `background` to `background-image`, which, in my opinion, should happen some day, we won't want to deprecate `n` blocks.

Filters or https://github.com/WordPress/gutenberg/blob/trunk/packages/blocks/src/api/parser/convert-legacy-block.js might required?

Reply:
The main issue is the `!important` on preset styles.

So, if an existing block has a preset gradient set, and a background image, we'd like to merge the two values into something like `background-image: var(--wp--preset--gradient-slug), url(...)` but the classname `has-slug-gradient-background` overwrites all this with `background: var(--wp--preset--gradient-slug) !important`.

So this deprecation strips the classname and classname attribute.

But then, when resetting the background image, the classname attribute need to be set back.

It makes me think we're going about this the wrong way. Maybe a new property in theme.json or something. I don't know.

Reply:
> It makes me think we're going about this the wrong way

It's this. 

My thinking is fuzzy. 

There is no deprecation. Blocks before the background-image migration will continue to work as they did before.

Only newly-added blocks will receive the _new_ feature. 

That means blocks that have already added a background image and a preset gradient will continue to work as they did before until they're edited, whereupon they'll be merged into one CSS property.

Deprecations are not required, so I'll remove this change.